src/openMVG/multiview/triangulation_nview.cpp

```diff
@@ -30,120 +30,127 @@
 
 namespace openMVG {
 
-  void TriangulateNView(const Mat2X &x,
-    const std::vector< Mat34 > &Ps,
-    Vec4 *X) {
-      Mat2X::Index nviews = x.cols();
-      assert(nviews == Ps.size());
-
-      Mat design = Mat::Zero(3*nviews, 4 + nviews);
-      for (int i = 0; i < nviews; i++) {
-        design.block<3, 4>(3*i, 0) = -Ps[i];
-        design(3*i + 0, 4 + i) = x(0, i);
-        design(3*i + 1, 4 + i) = x(1, i);
-        design(3*i + 2, 4 + i) = 1.0;
-      }
-      Vec X_and_alphas;
-      Nullspace(&design, &X_and_alphas);
-      *X = X_and_alphas.head(4);
+void TriangulateNView(const Mat2X &x,
+                      const std::vector< Mat34 > &Ps,
+                      Vec4 *X)
+{
+  Mat2X::Index nviews = x.cols();
+  assert(nviews == Ps.size());
+
+  Mat design = Mat::Zero(3 * nviews, 4 + nviews);
+  for(int i = 0; i < nviews; i++)
+  {
+    design.block<3, 4>(3 * i, 0) = -Ps[i];
+    design(3 * i + 0, 4 + i) = x(0, i);
+    design(3 * i + 1, 4 + i) = x(1, i);
+    design(3 * i + 2, 4 + i) = 1.0;
   }
-
-  typedef Eigen::Matrix<double, 2, 3> Mat23;
-  inline Mat23 SkewMatMinimal(const Vec2 &x) {
-    Mat23 skew;
-    skew <<
-      0, -1,  x(1),
-      1, 0, -x(0);
-    return skew;
+  Vec X_and_alphas;
+  Nullspace(&design, &X_and_alphas);
+  *X = X_and_alphas.head(4);
+}
+
+typedef Eigen::Matrix<double, 2, 3> Mat23;
+
+inline Mat23 SkewMatMinimal(const Vec2 &x)
+{
+  Mat23 skew;
+  skew <<
+          0, -1, x(1),
+          1, 0, -x(0);
+  return skew;
+}
+
+void TriangulateNViewAlgebraic(const Mat2X &x,
+                               const std::vector< Mat34 > &Ps,
+                               Vec4 *X)
+{
+  Mat2X::Index nviews = x.cols();
+  assert(nviews == Ps.size());
+
+  Mat design(2 * nviews, 4);
+  for(int i = 0; i < nviews; i++)
+  {
+    design.block<2, 4>(2 * i, 0) = SkewMatMinimal(x.col(i)) * Ps[i];
   }
+  Nullspace(&design, X);
+}
+
+double Triangulation::error(const Vec3 &X) const
+{
+  double squared_reproj_error = 0.0;
+  for(std::size_t i = 0; i < views.size(); i++)
+  {
+    const Mat34& PMat = views[i].first;
+    const Vec2 & xy = views[i].second;
+    const Vec2 p = Project(PMat, X);
+    squared_reproj_error += (xy - p).norm();
+  }
+  return squared_reproj_error;
+}
+
+// Camera triangulation using the iterated linear method
```

Comment:
documentation on functions should use doxygen formatting `///` or `/**` and should be done in header.